packages/react-dom/src/events/ReactBrowserEventEmitter.js

```diff
@@ -13,13 +13,31 @@ import {
   trapBubbledEvent,
   trapCapturedEvent,
 } from './ReactDOMEventListener';
+import {DOCUMENT_NODE, DOCUMENT_FRAGMENT_NODE} from '../shared/HTMLNodeType';
 import isEventSupported from './isEventSupported';
 import BrowserEventConstants from './BrowserEventConstants';
 
 export * from 'events/ReactEventEmitterMixin';
 
 var {topLevelTypes} = BrowserEventConstants;
 
+const forceDelegation = {
+  topMouseOver: 1,
+  topMouseOut: 2,
+  topMouseEnter: 3,
+  topMouseLeave: 4,
+};
```

Comment:
why do these need delegation? I wonder what the likelihood of this growing annoyingly is?

Reply:
We need this so that the EnterLeavePlugin event plugin works with elements React isn't watching.

Reply:
blah that's annoying, this should need to include `topMouseEnter` and `topMouseLeave`, at the moment we don't support the native event (we always polyfill it) and I think even if we did it wouldn't be needed, since those events always originate from the target element (react owned) whereas the other events originate from the relatedTarget, which may be not be react-owned

Reply:
also did we end up some place on using the native event vs continuing to polyfill this forever?

Reply:
Yeah. Still I can't help but wonder if there's work that could be done here to do the attachment local to the event plugin. It's a bummer that ReactBrowserEventEmitter and EnterLeavePlugin are coupled like this.

Reply:
This is a similar problem for change events on inputs. We have to attach change events to controlled inputs even if there isn't an event listener. This is so that controlled inputs that don't have `readOnly` prevent value changes on user input.

IMO, we should warn users when they don't have a change event and tell them to use `readOnly`, but we might already do that. Or, at the very least, we should attach the event listeners inside of the input wrappers (input, select, and textarea).

Reply:
> also did we end up some place on using the native event vs continuing to polyfill this forever?

Let's give it a look. It looks like we might be able to use the native event now, but the [support matrix for `mouseleave` on MDN is incomplete](https://developer.mozilla.org/en-US/docs/Web/Events/mouseleave).

We should do this research, kill the event plugin if we can, which I think would also let us get rid of some tree traversal methods specific to this event. 

At the very least we should update all of those question marks on MDN.

Reply:
I did the basic work checking for support, and we definitely can use the native event, the problem is Portals now. 

naive approach: https://github.com/facebook/react/pull/10247
hack with portal support: https://github.com/facebook/react/pull/10269

In terms of moving attachment to the plugin, there used to be (still are?) hooks on the plugins that would get called for this sort of thing, something like `willAttachListener` they may have been removed tho since i can't seem to find them...